internal/global/meter.go

```diff
@@ -152,14 +152,17 @@ func (m *meter) Int64Counter(name string, options ...metric.Int64CounterOption)
 		return m.delegate.Int64Counter(name, options...)
 	}
 
-	i := &siCounter{name: name, opts: options}
 	cfg := metric.NewInt64CounterConfig(options...)
 	id := instID{
 		name:        name,
-		kind:        reflect.TypeOf(i),
+		kind:        reflect.TypeOf(siCounter{}),
```

Comment:
nitpick: There would be a slight difference here as the previous implementation had pointer type in kind. You could use the following to ensure no changes. 
```suggestion
		kind:        reflect.TypeOf((*siCounter)(nil)),
```